vispy/visuals/colorbar.py

```diff
@@ -436,76 +314,125 @@ def _update(self):
             raise ValueError("half-height must be positive and non-zero"
                              ", not %s", halfh)
 
-        # Place the labels according to the given orientation
-        if self._orientation == "bottom":
-            text_x = x
-            text_y = y + _TEXT_PADDING_FACTOR * (halfh + self.border_width)
-
-            self._label.pos = text_x, text_y
-
-            # TODO, HACK: This should ideally be a single TextVisual
-            # However, one TextVisual with multiple strings
-            # does not seem to be working as of now. (See #981)
-            # https://github.com/vispy/vispy/issues/981
-            self._ticks[0].pos = x - halfw, text_y
-            self._ticks[1].pos = x + halfw, text_y
+        (label_pos, ticks_pos) = \
+            ColorBarVisual._calc_positions(center=self._center_pos,
```

Comment:
I created a static member which calculates the label positions. I prefer stuff that does pure computation to be static, since that reinforces the fact that it doesn't exactly depend on the class's state.